src/main/java/net/minecraftforge/client/ForgeHooksClient.java

```diff
@@ -989,6 +1001,68 @@ public static RenderType getEntityRenderType(RenderType chunkRenderType, boolean
         return RenderTypeHelper.getEntityRenderType(chunkRenderType, cull);
     }
 
+    public static IForgeGlslPreprocessor[] onShaderProgramCreation(GlslPreprocessor glslPreprocessor, ShaderInstance shaderInstance)
+    {
+        final RegisterGlslPreprocessorsEvent event = new RegisterGlslPreprocessorsEvent(shaderInstance);
+        ModLoader.get().postEvent(event);
+        final List<IForgeGlslPreprocessor> preprocessors = new ArrayList<>();
+        preprocessors.add(glslPreprocessor);
+        preprocessors.addAll(event.getPreprocessors());
+        return preprocessors.toArray(new IForgeGlslPreprocessor[0]);
+    }
+
+    public static void onLevelRenderingEnd(ProfilerFiller profilerfiller)
+    {
+        OITLevelRenderer.getInstance().endLevelRendering(profilerfiller);
+    }
+
+    public static void onLevelRenderingStart(ProfilerFiller profilerfiller)
+    {
+        OITLevelRenderer.getInstance().startLevelRendering(profilerfiller);
+    }
+
+    public static void renderChunkGeometry(RenderType renderType, PoseStack currentPoseStack, double cameraX, double cameraY, double cameraZ, Matrix4f projectionMatrix)
+    {
+        OITLevelRenderer.getInstance().checkHandlesThenQueueOrRender(renderType, currentPoseStack, cameraX, cameraY, cameraZ, projectionMatrix);
+    }
+
+    public static void renderParticles(ParticleRenderType particlerendertype, Iterable<Particle> particles, Frustum clippingHelper, Camera camera, float partialTickTime)
+    {
+        OITLevelRenderer.getInstance().checkHandlesThenQueueOrRender(particlerendertype, particles, clippingHelper, camera, partialTickTime);
+    }
+
+    public static void renderBuffer(RenderType renderType, BufferBuilder.RenderedBuffer buffer)
+    {
+        OITLevelRenderer.getInstance().checkHandlesThenQueueOrRender(renderType, buffer);
+    }
```

Comment:
Yes, on the rendering side we made the explicit decision to have all Patch hooks be in the ForgeHooksClient class, in case we need to modify the behaviour later.